### DISCUSSION TITLE
How to export/save changes after editing PDF document

### DISCUSSION THREAD

**Discussion by x80486** [JakubMelka/PDF4QT#233] [Q&A] [answered]
I recently found `PDF4QT`. I'm testing the software suite — at the moment the `1.4.0.0` (Flatpak) release.

I am experiencing something odd with `PDF4QT Editor`. I may be doing something wrong, but none of the changes I make to a given PDF document are reflected when I save the file.

For instance, I downloaded a random PDF document to redact some text and add an `EXPERIMENTAL` tag. This is how the document looks like in `PDF4QT Editor`:

![image](https://github.com/user-attachments/assets/bbf6f8cb-762b-4515-91be-3227561bcb10)

I can see exactly the same changes also when using the `PDF4QT Viewer` (from the suite) as well, but not a single PDF viewer, in GNU/Linux or macOS, recognizes the changes.

Could you please clarify which button or option is used to actually save the document after the same is edited/changed in `PDF4QT Editor`?

**Comment by JakubMelka** [comment, marked as answer]
Hello @x80486, if you are redacting the document, then this operation has two phases:

- mark text and graphics, which you want to redact
- **Create redacted document via tool button**. Please be careful when you are doing this, because the content will be deleted permanently from the PDF file.

**Comment by x80486** [comment]
All right. Thanks for the clarification. I didn't realize that option was actually on that menu.

Regardless, I think I may be doing something wrong, or perhaps the Flatpak release may have an issue with this, but I'm using [this](https://www.adobe.com/support/products/enterprise/knowledgecenter/media/c4611_sample_explain.pdf) Adobe PDF Sample file, and every time I try to modify something, for instance, using the redaction feature, when I finally select `Tools > Redact > Create Redacted Document`, the final output is a blank document.

**Reply by JakubMelka** [reply]
I have tried it on MS windows and it works. Could you please enable filesystem permission to the flatpak package?

**Reply by x80486** [reply]
I usually don't drag the discussions to another topic, so I'll open up a new one if I can't figure this one out, but I did enable `home` (and also `host`) access, and somehow the exported/created file only contains a few kilobytes — always around 11KB - 19KB based on the aforementioned sample file. Moreover, I don't see any errors or anything useful in the (journal) logs from the operating system.

Anyway, thank you so much for your help! :bowing_man:
